### PR TITLE
[tests-only][full-ci]Expand tests coverage for adding multiple members to a group at single request

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds tests coverage 
- Scenario Outline: try to a user-id with invalid characters to a group
- Scenario Outline: try to add user-ids with invalid characters to a group at once
- Scenario: add the same user to a group at once

## Related Issue
- [Fixes <issue_link>](https://github.com/owncloud/ocis/issues/5569)

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/TestHelpers/WebDavHelper.php [hunk 1]
```diff
@@ -435,6 +435,20 @@ public static function generateUUIDv4():string {
...
+	 * @throws Exception
+	 */
+	public static function generateInvalidUUID():string {
+		// generate 16 bytes (128 bits) of random data or use the data passed into the function.
+		$data = random_bytes(16);
+		\assert(\strlen($data) == 16);
+
+		return vsprintf('%s%s-%s-%s-%s-%s%s%s', str_split($data, 2));
+	}
+
```
Comment (SwikritiT) [inline tests/TestHelpers/WebDavHelper.php:451]: don't we already have a similar function?

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 2]
```diff
@@ -253,3 +253,27 @@ Feature: add users to group
     And user "Brian" has been added to group "music"
     When the administrator "Alice" tries to add a group "music" to a group "student" using the Graph API
     Then the HTTP status code should be "400"
+
+
+  Scenario: try to add multiple invalid character user-ids to a group at once
```
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:258]:
```suggestion
  Scenario: try to add multiple user-ids with invalid characters to a group at once
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 3]
```diff
@@ -253,3 +253,27 @@ Feature: add users to group
...
+    Given the administrator has given "Alice" the role "Admin" using the settings api
+    And user "Alice" has created a group "grp1" using the Graph API
+    When the administrator "Alice" tries to add user-ids with invalid characters to a group "grp1" at once using the Graph API
+    Then the HTTP status code should be "400"
+
+  @issue-5855
+  Scenario: add the same user to a group at once
```
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:265]:
```suggestion
  Scenario: add a user twice to a group at once
```

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 4]
```diff
@@ -1235,7 +1235,7 @@ public function theFollowingUsersShouldBeListedInFollowingGroups(TableNode $tabl
 		$this->featureContext->verifyTableNodeColumns($table, ['username', 'groupname']);
 		$usersGroups = $table->getColumnsHash();
 		foreach ($usersGroups as $userGroup) {
-			$members = $this->listGroupMembers($userGroup['groupname']);
+			$members = $this->featureContext->setResponse($this->listGroupMembers($userGroup['groupname']));
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/GraphContext.php:1238]: I don't know if we should set the response in the middle of a loop. Aren't there other alternatives for this?

File: tests/TestHelpers/WebDavHelper.php (hunk 1 above)
Comment (amrita-shrestha) [reply tests/TestHelpers/WebDavHelper.php:451]: i was for generateUUID

File: tests/acceptance/features/apiGraph/deleteUser.feature [hunk 5]
```diff
@@ -80,4 +80,3 @@ Feature: delete user
     When the user "Alice" deletes a user "Brian" using the Graph API
     Then the HTTP status code should be "204"
     And user "Brian" should not exist
-
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/deleteUser.feature:83]: You could do `git checkout origin/stable-2.0 tests/acceptance/features/apiGraph/deleteUser.feature` to undo this change

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 6]
```diff
@@ -253,3 +253,27 @@ Feature: add users to group
...
+    And user "Alice" has created a group "grp1" using the Graph API
+    When the administrator "Alice" adds the following users to a group "grp1" at once using the Graph API
+      | username |
+      | Brian    |
+      | Brian    |
+    Then the HTTP status code should be "204"
+    And the following users should be listed in the following groups
+      | username | groupname |
+      | Brian    | grp1      |
+    And the user "Brian" should be added "1" time in a group
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:279]: Are we gonna assert other than `1` in any case?
```suggestion
    And the user "Brian" should be added once to a group
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 7]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
...
     Then the HTTP status code should be "400"
+
+
+  Scenario Outline: try to add a user-id with invalid characters to a group
+    Given the administrator has given "Alice" the role "Admin" using the settings api
+    And user "Alice" has created a group "grp1" using the Graph API
+    When the administrator "Alice" tries to add a user-id "<UUID-with-invalid-character>" with invalid characters to a group "grp1" using the Graph API
+    Then the HTTP status code should be "404"
+    Examples:
+      | UUID-with-invalid-character           | comment                                                |
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:264]:
```suggestion
    When the administrator "Alice" tries to add a user-id "<invalid-uuidv4>" with invalid characters to a group "grp1" using the Graph API
    Then the HTTP status code should be "404"
    Examples:
      | invalid-uuidv4           | comment                                                |
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 8]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
...
+      | �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ               | UTF characters                                         |
+      | 4c510ada-c86b-4815-8820-42cdf82c3d511 | adding an extra character at end of valid UUID pattern |
+      | 4c510adac8-6b-4815-882042cdf-82c3d51  | invalid UUID pattern                                   |
+
+
+  Scenario Outline: try to user-ids with invalid characters to a group at once
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:270]:
```suggestion
  Scenario Outline: try to add invalid user ids to a group at once
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 9]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
...
+    When the administrator "Alice" tries to add the following user-id with an invalid character to a group "grp1" at once using the Graph API
+      | user-id                              |
+      | <first-UUID-with-invalid-character>  |
+      | <second-UUID-with-invalid-character> |
+    Then the HTTP status code should be "404"
+    Examples:
+      | first-UUID-with-invalid-character     | second-UUID-with-invalid-character    | comment                                                |
+      | �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ               | �ϰ@Ϧ-$ʨΌϖ-¶Ëøœ-@$@^ΌϖЁϿ               | UTF characters                                         |
+      | 4c510ada-c86b-4815-8820-42cdf82c3d511 | 4c5169da-d36b-aa15-88d0-42d4582c3d516 | adding an extra character at end of valid UUID pattern |
+      | 4c510adac8-6b-4815-882042cdf-82c3d51  | 4c5-169dad36b-aa1588-d042d45-82c3d51  | invalid UUID pattern                                   |
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:282]: Will this work?
```suggestion
    When the administrator "Alice" tries to add the following user-id with an invalid character to a group "grp1" at once using the Graph API
      | user-id          |
      | <invalid-uuidv4> |
      | <invalid-uuidv4> |
    Then the HTTP status code should be "404"
    Examples:
      | invalid-uuidv4                        | comment                                                |
      | �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ               |  UTF characters                                         |
      | 4c510ada-c86b-4815-8820-42cdf82c3d511 | adding an extra character at end of valid UUID pattern |
      | 4c510adac8-6b-4815-882042cdf-82c3d51  | invalid UUID pattern                                   |
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 10]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
     And user "Brian" has been added to group "music"
     When the administrator "Alice" tries to add a group "music" to a group "student" using the Graph API
     Then the HTTP status code should be "400"
+
+
+  Scenario Outline: try to add a user-id with invalid characters to a group
+    Given the administrator has given "Alice" the role "Admin" using the settings api
+    And user "Alice" has created a group "grp1" using the Graph API
+    When the administrator "Alice" tries to add a user-id "<UUID-with-invalid-character>" with invalid characters to a group "grp1" using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:261]:
```suggestion
    When the administrator "Alice" tries to add an invalid user-id "<UUID-with-invalid-character>" to a group "grp1" using the Graph API
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 11]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
...
+      | UUID-with-invalid-character           | comment                                                |
+      | �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ               | UTF characters                                         |
+      | 4c510ada-c86b-4815-8820-42cdf82c3d511 | adding an extra character at end of valid UUID pattern |
+      | 4c510adac8-6b-4815-882042cdf-82c3d51  | invalid UUID pattern                                   |
+
+
+  Scenario Outline: try to user-ids with invalid characters to a group at once
+    Given the administrator has given "Alice" the role "Admin" using the settings api
+    And user "Alice" has created a group "grp1" using the Graph API
+    When the administrator "Alice" tries to add the following user-id with an invalid character to a group "grp1" at once using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:273]:
```suggestion
    When the administrator "Alice" tries to add the following invalid user-ids to a group "grp1" at once using the Graph API
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 12]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
     And user "Brian" has been added to group "music"
     When the administrator "Alice" tries to add a group "music" to a group "student" using the Graph API
     Then the HTTP status code should be "400"
+
+
+  Scenario Outline: try to add invalid user id to a group
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:258]:
```suggestion
  Scenario Outline: try to add user with invalid id to a group
```

Comment (amrita-shrestha) [reply tests/acceptance/features/apiGraph/addUserToGroup.feature:258]: https://github.com/owncloud/ocis/pull/5864#discussion_r1144210941

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 13]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
...
+    When the administrator "Alice" tries to add the following invalid user ids to a group "grp1" at once using the Graph API
+      | user-id        |
+      | invalid-uuidv4 |
+      | invalid-uuidv4 |
+    Then the HTTP status code should be "404"
+    Examples:
+      | invalid-uuidv4                        | comment                                                |
+      | �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ               | UTF characters                                         |
+      | 4c510ada-c86b-4815-8820-42cdf82c3d511 | adding an extra character at end of valid UUID pattern |
+      | 4c510adac8-6b-4815-882042cdf-82c3d51  | invalid UUID pattern                                   |
```
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:282]: Should this be like this?
```suggestion
  Scenario: try to add invalid user ids to a group at once
    Given the administrator has given "Alice" the role "Admin" using the settings api
    And user "Alice" has created a group "grp1" using the Graph API
    When the administrator "Alice" tries to add the following invalid user ids to a group "grp1" at once using the Graph API
      | user-id        |
      |  �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ  |
      | 4c510ada-c86b-4815-8820-42cdf82c3d511 |
    | 4c510adac8-6b-4815-882042cdf-82c3d51 |
     Then the HTTP status code should be "404"
```

File: tests/acceptance/features/apiGraph/addUserToGroup.feature (hunk 12 above)
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/addUserToGroup.feature:258]: A user cannot have an invalid id and `add user with invalid id` could lead to misunderstanding that there can be a valid user with invalid id pattern

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 14]
```diff
@@ -253,3 +253,44 @@ Feature: add users to group
...
+    When the administrator "Alice" tries to add an invalid user id "<invalid-uuidv4>" to a group "grp1" using the Graph API
+    Then the HTTP status code should be "404"
+    Examples:
+      | invalid-uuidv4                        | comment                                                |
+      | �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ               | UTF characters                                         |
+      | 4c510ada-c86b-4815-8820-42cdf82c3d511 | adding an extra character at end of valid UUID pattern |
+      | 4c510adac8-6b-4815-882042cdf-82c3d51  | invalid UUID pattern                                   |
+
+
+  Scenario Outline: try to add invalid user ids to a group at once
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/addUserToGroup.feature:270]: A user cannot have an invalid id and `add user with invalid id` could lead to a misunderstanding that there can be a valid user with an invalid id pattern

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 15]
```diff
@@ -1253,6 +1253,31 @@ public function theFollowingUsersShouldBeListedInFollowingGroups(TableNode $tabl
...
+
+		foreach ($members as $member) {
+			if ($member['onPremisesSamAccountName'] === $user) {
+				$count++;
+			}
+		}
+		Assert::assertEquals(
+			1,
+			$count,
+			"The user '" . $user . "' has been added '" . $count . "' time in a group. "
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/GraphContext.php:1277]:
```suggestion
			"Expected count of user  " .  $user . "in a group " . $group . " to be 1 but found " . $count 
```
may be something like this ?

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 16]
```diff
@@ -2282,4 +2282,86 @@ public function userTriesToAddTheFollowingUsersToAGroupAtOnceWithInvalidJsonUsin
...
+
+		foreach ($members as $member) {
+			if ($member['onPremisesSamAccountName'] === $user) {
+				$count++;
+			}
+		}
+		Assert::assertEquals(
+			1,
+			$count,
+			"Expected the user to be added to a group once but found '" . $count . "' time"
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/GraphContext.php:2364]: Others look good :+1: 
```suggestion
			"Expected user '" . $user . "' to be added once to group '" . $group . "' but the user is listed '" . $count . "' times"
```

**Review by saw-jan — APPROVED** [review]
LGTM

File: tests/acceptance/features/apiGraph/addUserToGroup.feature [hunk 17]
```diff
@@ -311,3 +311,44 @@ Feature: add users to group
...
+      | <invalid-uuidv4> |
+      | <invalid-uuidv4> |
+    Then the HTTP status code should be "404"
+    Examples:
+      | invalid-uuidv4                        | comment                                                |
+      | �ϰ�Ϧ-@$@^-¶Ëøœ-ɧɸɱʨΌϖЁϿ               | UTF characters                                         |
+      | 4c510ada-c86b-4815-8820-42cdf82c3d511 | adding an extra character at end of valid UUID pattern |
+      | 4c510adac8-6b-4815-882042cdf-82c3d51  | invalid UUID pattern                                   |
+
+  @issue-5855
```
Comment (ScharfViktor) [inline tests/acceptance/features/apiGraph/addUserToGroup.feature:342]: issue is closed. I guest that it works in the master?
if yes, could you put tag`@skipOnStable2.0` and remove bug from expected failures list

**Review by grgprarup — APPROVED** [review]
LGTM :+1:
